src/test/java/org/zalando/nakadi/utils/TestUtils.java

```diff
@@ -179,11 +176,15 @@ public static void waitFor(final Runnable runnable, final long timeoutMs) {
     }
 
     @SuppressWarnings("unchecked")
-    public static void waitFor(final Runnable runnable, final long timeoutMs, final int intervalMs) {
+    public static void waitFor(final Runnable runnable, final long timeoutMs, final int intervalMs,
+                               final Class<? extends  Throwable>... additionalException) {
+        final Class<? extends Throwable>[] leadToRetry = new Class[additionalException.length + 1];
+        System.arraycopy(additionalException, 0, leadToRetry, 1, additionalException.length);
+        leadToRetry[0] = AssertionError.class;
```

Comment:
You are array lover :)
I think it can be done much easier:
```java
final List<Class<? extends Throwable>> exs = Arrays.asList(additionalException);
exs.add(AssertionError.class);
```